def summa(a,b):
    """Данная функция выполняет математическую операцию, сложения.
        Аргументы получаеммые функцией на вход:
        a - string, первое число для математической операции.
        b - string, второе число для математической операции.
        Исключения прорабатываемые функцией:
        ValueError - функция получает аргумент правильного типа, но некорректного значения.
        TypeError - операция применена к объекту несоответствующего типа. 
        OverflowError - возникает, когда результат арифметической операции слишком велик для представления.
        В результате выполнения функции выводится результат сложения двух чисел.
    """
    try: 
        int(a)
    except ValueError:
        print("Введите первое число еще раз")
    else:
        a = int(a)
    try: 
        int(b)
    except ValueError:
        print("Введите второе число еще раз")
    else:
        b = int(b)
    try:
        a+b
    except TypeError:
        print("Ошибка с типами переменных")
    except OverflowError:
        print("Возникло переполнение")
    else:
        print("Сумма двух чисел равна", a+b)


def razn(a,b):
    """Данная функция выполняет математическую операцию, вычитание.
        Аргументы получаеммые функцией на вход:
        a - string, первое число для математической операции.
        b - string, второе число для математической операции.
        Исключения прорабатываемые функцией:
        ValueError - функция получает аргумент правильного типа, но некорректного значения.
        TypeError - операция применена к объекту несоответствующего типа. 
        OverflowError - возникает, когда результат арифметической операции слишком велик для представления.
        В результате выполнения функции выводится результат вычитание одного числа из другого.
    """
    try: 
        int(a)
    except ValueError:
        print("Введите первое число еще раз")
    else:
        a = int(a)
    try: 
        int(b)
    except ValueError:
        print("Введите второе число еще раз")
    else:
        b = int(b)
    try:
        a-b
    except TypeError:
        print("Ошибка с типами переменных")
    except OverflowError:
        print("Возникло переполнение")
    else:
        print("Разница двух чисел равна", a-b)


def proz(a,b):
    """Данная функция выполняет математическую операцию, умножение.
        Аргументы получаеммые функцией на вход:
        a - string, первое число для математической операции.
        b - string, второе число для математической операции.
        Исключения прорабатываемые функцией:
        ValueError - функция получает аргумент правильного типа, но некорректного значения.
        TypeError - операция применена к объекту несоответствующего типа. 
        OverflowError - возникает, когда результат арифметической операции слишком велик для представления.
        В результате выполнения функции выводится результат перемножения двух чисел.
    """
    try: 
        int(a)
    except ValueError:
        print("Введите первое число еще раз")
    else:
        a = int(a)
    try: 
        int(b)
    except ValueError:
        print("Введите второе число еще раз")
    else:
        b = int(b)
    try:
        a*b
    except TypeError:
        print("Ошибка с типами переменных")
    except OverflowError:
        print("Возникло переполнение")
    else:
        print("Произведение двух чисел равна", a*b)


def delet(a,b):
    """Данная функция выполняет математическую операцию, деление.
        Аргументы получаеммые функцией на вход:
        a - string, первое число для математической операции.
        b - string, второе число для математической операции.
        Исключения прорабатываемые функцией:
        ValueError - функция получает аргумент правильного типа, но некорректного значения.
        TypeError - операция применена к объекту несоответствующего типа. 
        OverflowError - возникает, когда результат арифметической операции слишком велик для представления.
        ZeroDivisionError - деление на ноль.
        В результате выполнения функции выводится результат деления одного числа на другое.
    """
    try: 
        int(a)
    except ValueError:
        print("Введите первое число еще раз")
    else:
        a = int(a)
    try: 
        int(b)
    except ValueError:
        print("Введите второе число еще раз")
    else:
        b = int(b)
    try:
        a / b
    except TypeError:
        print("Ошибка с типами переменных")
    except ZeroDivisionError:
        print("Делить на ноль нельзя")
    except OverflowError:
        print("Возникло переполнение")
    else:
        print("Деление превого числа на второе равно", a/b)


def step(a,b):
    """Данная функция выполняет математическую операцию, возведение в степень.
        Аргументы получаеммые функцией на вход:
        a - string, первое число для математической операции.
        b - string, второе число для математической операции.
        Исключения прорабатываемые функцией:
        ValueError - функция получает аргумент правильного типа, но некорректного значения.
        TypeError - операция применена к объекту несоответствующего типа. 
        OverflowError - возникает, когда результат арифметической операции слишком велик для представления.
        В результате выполнения функции выводится результат возведения в степень.
    """
    try: 
        int(a)
    except ValueError:
        print("Введите первое число еще раз")
    else:
        a = int(a)
    try: 
        int(b)
    except ValueError:
        print("Введите второе число еще раз")
    else:
        b = int(b) 
    try:
        a ** b
    except TypeError:
        print("Ошибка с типами переменных")
    except OverflowError:
        print("Возникло переполнение")
    else:
        print("Произведение двух чисел равна", a**b)
        
              
while(True):  # Запускается вечный цикл
    print("Введите первое число в калькулятор")
    a = input()
    print("Введите команду представленную из списка")
    print("Для суммирования чисел введите +")
    print("Для вычисления разности чисел введите -")
    print("Для перемножения чисел введите *")
    print("Для вычисления деления чисел введите /")
    print("Для возведения в степень одного из чисел введите ^")
    opr = input()
    print("Введите второе число в калькулятор")
    b = input()
    if opr == '+':  # Проверяем какой оператор математического действия был введен
        summa(a,b)
    if opr == '-':
        razn(a,b)
    if opr == '*':
        proz(a,b)
    if opr == '/':
        delet(a,b)
    if opr == '^':
        step(a,b)
    print("Желаете продолжить работу с калькулятором ? Для продолжения работы введите Да")
    st = input()
    if st == '':  # Проверяем можем ли мы выходить из цикла при пустой строке послк ввода
        break